src/cron/tasks/monthly_export.ts

```diff
@@ -99,59 +101,68 @@ const task: Task = {
             const ele = parseFloat(dataArr[index++]);
             if (ele !== 0) content[key] = ele;
           });
+          if (!disableDataLoss && option.groupTimeRange === 'month' && content['2021-10']) {
+            // handle data loss in 2021.10 only when generate data by month
+            content['2021-10-raw'] = content['2021-10'];
+            const arr = ['2021-08', '2021-09', '2021-11', '2021-12'].map(m => content[m]);
+            if (!arr[3]) arr[3] = 0;
+            if (!arr[2]) arr[2] = 0;
+            if (!arr[0]) arr[0] = arr[3];
+            if (!arr[1]) arr[1] = arr[2];
+            content['2021-10'] = Math.round([0.15, 0.35, 0.35, 0.15].map((f, i) => f * arr[i]).reduce((p, c) => p + c));
+          }
```

Comment:
The process will only take effect if 202110 data point exists and only for statistical metric when group by month.

The basic logic for the process is using `0.15 * Value_202108 + 0.35 * Value_202109 + 0.35 * Value_202111 + 0.15 * Value_202112`.

But since some repo may just occur after 2021.10, so replace with 202112 if 202108 not exists and replace with 202111 if 202109 not exists.

And for all the metrics, use integer format rather than float format because some metrics requires to be an integer.

Reply:
Hi @frank-zsy, I think it can be good if you also put details in this review comment as a code comment so possible future matainers are able to understand the code detail with ease and quickly :-)

Reply:
Thanks, good point, I will add more code comment in the future.